### PR TITLE
core: delete entry from txs after adding correct sender

### PR DESCRIPTION
**Pull request by kimmylin**
when `from` is different from the signing account, remove the old entry when the new one is added to the transaction set.

### CODE REVIEW

**Comment by holiman** [comment]
Could you provide some more context for this? Just looking at the fix, I can't really see where this is used and what situation will trigger buggy behaviour (and similarly -- what potential sideeffects the fix may have)

**Comment by kimmylin** [comment]
thanks. i'm working on an unrelated change (the go implementation of Casper from the AMIS team), came across this code and thought this may be problematic. i was a C++/Java programmer and just started working with go recently.

based on https://github.com/ethereum/go-ethereum/pull/14773, there is an corner case where the actual signer is different from the one cached in the tx. when that's the case, won't we have two keys mapped to almost identical list of tx in `txs`? 

i am not well-versed enough of this codebase to assess the potential side effect tho. sorry.

**Comment by karalabe** [comment]
Oh wow, this PR took me 3 looks on 3 separate occasions to finally understand it.Yes, if the signer changes, the existing code is invalid as transactions would be duplicated.

The only time though when this could happen is during hard fork transitions that also modify the signing logic. Until now we had 2 such cases: Homestead disabled half of the frontier signatures for security reasons, and EIP155 introduced chain IDs.

TL;DR: This code path is probably a no-event for mainnet and any uptodate chains, but you are right that it should be fixed.
